feat: Add Migrations and Seed for Captions

zxkevin opened the pull request:
Added the migration for creation of captions table
and added Caption seed data

#60

zxkevin committed:
feat: Add Migrations and Seed for Captions

Added the migration for creation of captions table
and added Caption seed data

#60

zxkevin committed:
feat: Added Migration for users table

Added the migration for creation of users table
with username, first_name, last_name fields

#60

zxkevin committed:
fix: Disable field in user, notnullable to FK

Added a boolean field to users table on creation
Added not nullable condition to the FK post_id field in captions

#60

3dxp commented on packages/hackmcx-api/src/db/migrations/2022-03-08_1129_create_table_captions.cjs:
```diff
@@ -0,0 +1,21 @@
+            table.string('caption', 2048).notNullable();
+            table.integer('post_id').unsigned();
+            table.foreign('post_id').references('id').inTable('posts').onDelete('cascade');
+            table.float('average_rating',2,1);
```
We just want to grab the defaults here, it should provide less compatibility issues. 

```suggestion
            table.float('acc_rating');
```

3dxp commented:
Nice work on this @zxkevin. Looks good to me. Should be good to merge after comments are addressed.

zxkevin committed:
fix: Changed data type of average_rating

Changed data type of average_rating to double from float and
removed number formatting

#60

acampb65 commented:
Looks good to me!

acampb65 commented:
We probably want to add the notNullable() method to the username, first_name, and last_name columns of the users table during migration.

acampb65 commented:
`2022-03-09_1502.cjs` should probably be renamed to something like `2022-03-09_1502_demo_captions.cjs`

acampb65 committed:
fix: Rename Seed Data File

Changed the seed data filename in order to better understand what the file does.

zxkevin commented:
> 

I left the name fields nullable for maybe the anonymous users in the future and just to give people the option to decide if or not they want to include their name. The username is notnullable by default due to it being the primary key.

acampb65 committed:
fix: Add notnullable and imageUrl to User Table

Added a column for image URLs to the users table
Added not nullable condition to username, first_name, last_name, and
imageUrl

acampb65 committed:
feat: Add Seed Data for Users

Added seed data for the users table

acampb65 committed:
fix: Make Names Nullable, Add Default Image

Removed notNullable condition from first and last name.
Added default value for images.

acampb65 reviewed (approved):
Looks good
